### PR TITLE
Remove state.State.ForModel

### PR DESCRIPTION
**Pull request by axw**
## Description of change

Drop State.ForModel in favour of using StatePool everywhere.

## QA steps

Run tests, smoke test bootstrap/deploy/destroy-controller.

## Documentation changes

None.

## Bug reference

None.

### CODE REVIEW

**Comment by axw** [comment]
[Get rid of State.ForModel](https://trello.com/c/x6zW4zte/25-get-rid-of-stateformodel)

File: scripts/juju-force-upgrade/main.go [hunk 1]
```diff
@@ -58,7 +58,7 @@ func getState() (*state.State, error) {
 	st, err := state.Open(state.OpenParams{
 		Clock:              clock.WallClock,
 		ControllerTag:      config.Controller(),
-		ControllerModelTag: config.Model(),
+		ControllerModelTag: names.NewModelTag(modelUUID), // XXX field misnomer
```
Comment (wallyworld) [inline scripts/juju-force-upgrade/main.go]: really? we shouldn't be using ControllerModel tag?

Comment (axw) [reply scripts/juju-force-upgrade/main.go]: that's a mistake that I realised later on in the PR, just failed to clean up - thanks, will fix

I do think the OpenParams requires too much to be passed in though. We should be able to figure out the controller and controller model tags from the database. The caller should only have to pass in a model tag (to open a specific model's State) at most. There's no good reason to treat the controller model specially, AFAICT.

**Comment by axw** [comment]
$$merge$$

**Pull request merged**